hazelcast/src/test/java/com/hazelcast/osgi/HazelcastOSGiServiceTest.java

```diff
@@ -347,6 +510,30 @@ public void instanceShutdownSuccessfully() {
     }
 
     @Test
+    public void instanceShutdownSuccessfullyAlthoughThereIsExceptionWhileDeregister() {
+        final String INSTANCE_NAME = "test-osgi-instance";
+
+        registerDeregisterListener.throwExceptionOnDeregister = true;
+
+        HazelcastInternalOSGiService service = getService();
+
+        Config config = new Config(INSTANCE_NAME);
+
+        HazelcastOSGiInstance osgiInstance = service.newHazelcastInstance(config);
+        assertTrue(osgiInstance.getLifecycleService().isRunning());
+
+        HazelcastInstance instance = osgiInstance.getDelegatedInstance();
+        assertTrue(instance.getLifecycleService().isRunning());
```

Comment:
This should not be here. If you don't believe that your instance is running, you should make a test for that explicitly. 

This test should only keep the last 4 asserts.